collections/_posts/2024-03-20-ImageBind.md

```diff
@@ -0,0 +1,121 @@
+
+Each modality $$M$$ has its own dedicated encoder (see below for details) and an associated projector $$f_\mathcal{M}$$, so that every modality embedding ends up in a space of common dimension.
+
+The model is trained with the standard InfoNCE contrastive loss. Given a batch of observations, it aims to make the embeddings of a pair ($$\mathcal{I}_i$$, $$\mathcal{M}_i$$) closer while pulling away the 'negative' observations, that is all the other pairs of data in the batch. The loss can be written :
+
+$$
+\mathcal{L}_{\mathcal{I},\mathcal{M}} = - \log \frac{\exp(\textbf{q}_i^T \textbf{k}_i / \tau)}{\exp(\textbf{q}_i^T \textbf{k}_i / \tau) + \sum_{i\neq j} \exp(\textbf{q}_i^T \textbf{k}_j/\tau)}
+$$
+
+&nbsp;
+
+## Modality encoders details
+
+* **Image and video :** Vision Transformer (ViT). For video , they use 2-frame clips, with a video patch of size $$2 \times 16 \times 16$$ ($$T\times H \times W$$). Encode's weights are inflated to work with spatiotemporal patches and at inference, features from successive 2-frame clips are aggregated. They leverage image-text supervision from large-scale data by using OpenCLIP's image encoder weights (ViT-H, 630M parameters). The weights are kept frozen.
+
+* **Audio :** Audio samples are recorded at 16kHz. They are cut into 2-second samples and converted into spectrograms using 128-mel spectrograms bins. Spectrograms are processed as images with ViT with a patch size of 16 and stride 10.
+
+* **Text :** They directly use CLIP encoder architecture, which is a VIT. Again, Tyey leverage image-text supervision from large-scale data by using OpenCLIP's text encoder weights (320M parameters). The weights are kept frozen.
+
+* **Depth and Thermal data :** They treat both as one-channel images and use ViT to encoder them. 
+
+* **IMU :** The 6-channel signal acquisition is cut into 5-second clips, giving 2K time steps IMU. They are projected using 1D convolution with kernel size of 8 and then encoded using a Transformer.
+
+&nbsp;
+
+# Experiments
+
+## Datasets
+
+The authors use several dataset to couple audio, depth maps, thermal data and IMU to image (or video). Each dataset is split into train/test to evaluate zero-shot classification or retrieval abilities.
+
+<div style="text-align:center">
+<img src="/collections/images/imagebind/databases.jpg" width=500></div>
+<p style="text-align: center;font-style:italic">Figure 2. Overview of datasets used.</p>
+
+## Emergent zero-shot classification
+
+Zero-shot is a term used for model trained with a modality X (image or audio for instance) paired with text. The class is determined using simple standard prompts, such as "An image of a {class}", where {class} is replaced by every possible class. The class attributed to a sample is the one for with the associated text prompt has the highest similarity with the sample.  
```

Comment:
"Zero-shot is a term used for model trained with a modality X (image or audio for instance) paired with text". I do not understand what you mean here. 

My understanding would be 
" Zero-shot is a term used for a model trained with a modality pair X-Y (image or audio for instance) paired with text and applied to a modality pair Y-Z during inference."

Reply:
It was not clear ideed, i've reworded this paragraph